charactersheet/charactersheet/viewmodels/dm/encounter_sections/player_text_section.js

```diff
@@ -140,6 +144,17 @@ function PlayerTextSectionViewModel(parentEncounter) {
 
     self.modalFinishedClosing = function() {
         self.selectPreviewTab();
+
+        if (self.openModal()) {
+            self.playerTexts().forEach(function(item, idx, _) {
```

Comment:
This code repeats itself multiple times in this PR and it's a little verbose. I'd rather it be abstracted to the VM utilities or something. 

```js
self.playerTexts.updateItem(index, newData);

// KO...
ko.observableArray.prototype.updateItem = function(idx, newItem) {
  // Your code...
}
```

This would make all of this less error prone, and much easier to read. You could even include one that takes a callback with the found item for monsters.

```js
// KO...
ko.observableArray.prototype.updateItem = function(idx, newItem, onfind) {
  // Your code...
}

// Monsters
self.monsters.updateItem(index, newData, function(monster) {
  // Update ability scores.
});

```